dataset_example.py

```diff
@@ -0,0 +1,161 @@
+import typing
+from abc import ABC
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+
+from flytekit import kwtypes, task, workflow
+from flytekit.types.structured.structured_dataset import (
+    FLYTE_DATASET_TRANSFORMER,
+    DatasetDecodingHandler,
+    DatasetEncodingHandler,
+    DatasetFormat,
+    StructuredDataset,
+)
+
+PANDAS_PATH = "/tmp/pandas.pq"
+NUMPY_PATH = "/tmp/numpy.pq"
+BQ_PATH = "bq://photo-313016:flyte.new_table3"
+
+# https://github.com/flyteorg/flyte/issues/523
+# We should also support map and list in schema column
+column_type = kwtypes(w=typing.Dict[str, typing.Dict[str, int]], x=typing.List[typing.List[int]], y=int, z=str)
```

Comment:
```suggestion
my_cols = kwtypes(w=typing.Dict[str, typing.Dict[str, int]], x=typing.List[typing.List[int]], y=int, z=str)
```
just to make it clear